README.md

```diff
@@ -82,18 +83,36 @@ $ mkdir -p /data/near-lake-custom && minio server /data
 - add `s3_endpoint` parameter to LakeConfig instance
 
 ```bash
-let config = LakeConfig {
-    s3_endpoint: "http://0.0.0.0:9000".to_string(), // AWS S3 custom API endpoint
-    s3_bucket_name: "near-lake-custom".to_string(), // AWS S3 bucket name
-    s3_region_name: "eu-central-1".to_string(), // AWS S3 bucket region
-    start_block_height: 1, // the latest block height
-};
+let config = LakeConfigBuilder::default()
+        .s3_bucket_name("near-lake-custom")
+        .start_block_height(1)
+        .with_custom_endpoint("http://0.0.0.0:9000")
+        .build()
+        .expect("Failed to build Lakeconfig");
 ```
 
 ### AWS S3 Credentials
 
 In order to be able to get objects from the AWS S3 bucket you need to provide the AWS credentials.
 
+#### Passing credentials to the config builder
+
+```rust
+let config = LakeConfigBuilder::default()
+ .s3_bucket_name("near-lake-testnet")
+ .s3_region_name("eu-central-1")
+ .start_block_height(82422587)
+ .with_custom_credentials(
+      "AKIAIOSFODNN7EXAMPLE",
+      "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
+  )
+ .build()
+ .expect("Failed to build LakeConfig");
```

Comment:
I don't like the idea. My intent was to avoid users setting up their SdkConfig. What if we completely change our backend in the future?

Do you insist? We can discuss it now or later.

Reply:
> What if we completely change our backend in the future?

In that case with the current approach you will need to change a ton of different fields in LakeConfig instead of changing just one (s3_config).

Let's see how DevConsole integration could look like, and that might inform us on the configuration options here. Can you sketch it out maybe?

Reply:
Let's try to sketch.

What I want to change is the passing of the s3 bucket name a little bit.

I would like to have

```rust
let config = LakeConfigBuilder::default()
    .mainnet()
    .custom_credentials("ACCESSID", "SECRET")
    .start_block_height(1)
    .build();
```

And `.testnet()` for testnet. These would be "predefined" variants.

For self-hosted/custom environment with S3-compatible APIs this might look like

```rust
let config = LakeConfigBuilder::default()
    .custom_endpoint("http://0.0.0.0:9000/")
    .custom_bucket_name("near-lake-data-custom")
    .start_block_height(1)
    .build();
```

Talking about DevConsole integration, I would expect something like this (providing the credentials):

```rust
let config = LakeConfigBuilder::default()
    .pagoda_key("IAMPAGODAAPIKEY")
    .mainnet()
    .start_block_height(1)
    .build();
```

Reply:
With Pagoda we can make it as short as this:

```rust
let config = LakeConfigBuilder::default()
    .mainnet("I_AM_PAGODA_API_KEY")
    .start_block_height(1)
    .build();
```

With custom S3 storage:

```rust
let s3_config = aws_sdk::builder()
    .region("eu-central-1")
    .build();
let config = LakeConfigBuilder::default()
    .s3_config(s3_config)
    .s3_bucket_name("near-lake-data-custom")
    .start_block_height(1)
    .build();
```

If one would want to customize their S3 connection config, they can be free to do so with adjusting `s3_config`:

```rust
let s3_config = aws_sdk::builder()
    .endpoint_resolver(
        Endpoint::immutable(Uri::from_static("http://localhost:8080"))
    )
    .build();
```

I do not insist, we can land it as is now, but we may want to revisit it later.

Reply:
Fair enough. Do you want me to perform these changes in this PR for 0.3.0 version?

Reply:
I think we can revisit it when we will introduce DevConsole integration. Though, to minimize the disruption, I would strongly consider using `s3_config` from the get go, so instead of later cutting 0.4.0 with another set of breaking changes, we would be able to cut 0.3.1, and those who previously used 0.3.0 with direct AWS setup won't need to fear upgrading